src/main/java/classify/data/DataReader.java

```diff
@@ -76,6 +77,23 @@ public static void restoreStudentList(String studentFileInput, ArrayList <Studen
 
             //Set remarks
             student.getAttributes().setRemarks(inputArr[4].trim());
+
+            //Get all Subject Info
+            String [] allSubjects = (inputArr[5].trim()).split("#--#");
+            int numberOfSubjects = allSubjects.length;
+
+            for (int i = 0; i < numberOfSubjects ; i++){
+                String [] subjectDetailedInfo = allSubjects[i].split("##");
```

Comment:
Ensure to ban this use of character in inputparsing, and update it accordingly in UG. Otherwise, awesome job!